gluten-uniffle/velox/src/main/java/org/apache/spark/shuffle/gluten/uniffle/GlutenRssShuffleManager.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.spark.executor.ShuffleWriteMetrics;
+import org.apache.spark.shuffle.ColumnarShuffleDependency;
+import org.apache.spark.shuffle.RssShuffleHandle;
+import org.apache.spark.shuffle.RssShuffleManager;
+import org.apache.spark.shuffle.RssSparkConfig;
+import org.apache.spark.shuffle.ShuffleHandle;
+import org.apache.spark.shuffle.ShuffleWriteMetricsReporter;
+import org.apache.spark.shuffle.ShuffleWriter;
+import org.apache.spark.shuffle.sort.ColumnarShuffleManager;
+import org.apache.spark.shuffle.writer.VeloxUniffleColumnarShuffleWriter;
+import org.apache.uniffle.common.exception.RssException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Constructor;
+
+public class GlutenRssShuffleManager extends RssShuffleManager {
+  private static final Logger LOG = LoggerFactory.getLogger(GlutenRssShuffleManager.class);
+  private static final String GLUTEN_SHUFFLE_MANAGER_NAME =
+      "org.apache.spark.shuffle.sort.ColumnarShuffleManager";
+
+  private static final String VANILLA_UNIFFLE_SHUFFLE_MANAGER_NAME =
+      "org.apache.spark.shuffle.celeborn.SparkShuffleManager";
+  private volatile ColumnarShuffleManager _columnarShuffleManager;
+  private volatile RssShuffleManager _vanillaUniffleShuffleManager;
+
+  private ColumnarShuffleManager columnarShuffleManager() {
+    if (_columnarShuffleManager == null) {
+      synchronized (this) {
+        if (_columnarShuffleManager == null) {
+          _columnarShuffleManager =
+              initShuffleManager(GLUTEN_SHUFFLE_MANAGER_NAME, sparkConf, isDriver());
+        }
+      }
+    }
+    return _columnarShuffleManager;
+  }
+
+  private RssShuffleManager vanillaUniffleShuffleManager() {
+    if (_vanillaUniffleShuffleManager == null) {
+      synchronized (this) {
+        if (_vanillaUniffleShuffleManager == null) {
+          initShuffleManager(VANILLA_UNIFFLE_SHUFFLE_MANAGER_NAME, sparkConf, isDriver());
+        }
+      }
+    }
+    return _vanillaUniffleShuffleManager;
+  }
+
+  private boolean isDriver() {
+    return "driver".equals(SparkEnv.get().executorId());
+  }
+
+  private ColumnarShuffleManager initShuffleManager(String name, SparkConf conf, boolean isDriver) {
+    Constructor constructor;
+    ColumnarShuffleManager instance;
+    try {
+      Class klass = Class.forName(name);
+      try {
+        constructor = klass.getConstructor(conf.getClass(), Boolean.TYPE);
+        instance = (ColumnarShuffleManager) constructor.newInstance(conf, isDriver);
+      } catch (NoSuchMethodException var7) {
+        constructor = klass.getConstructor(conf.getClass());
+        instance = (ColumnarShuffleManager) constructor.newInstance(conf);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("initColumnManager fail");
+    }
+    return instance;
+  }
+
+  public GlutenRssShuffleManager(SparkConf conf, boolean isDriver) {
+    super(conf, isDriver);
+    conf.set("spark.rss.row.based", "false");
```

Comment:
Nit: it can have potential issue to set with raw string. Maybe, better to set it like this:
https://github.com/apache/incubator-gluten/pull/3767/files#diff-c5aeb43f0c96e20fabd596e8d6b8545669ce6bb966ca797f0041b2c3889957aaR112